assembl/tests/views_tests/test_view_auth.py

```diff
@@ -217,3 +218,53 @@ def test_merge_social_account(
     assert account.uid == p1_uid
     assert account.profile == participant1_user
     account.delete()
+
+
+def test_autologin(
+        test_session, test_app_participant1, closed_discussion, participant1_user,
+        participant1_social_account, google_identity_provider, request,
+        test_participant1_webrequest):
+    # as a logged in participant, I can see the discussion
+    long_ago = datetime(2000, 1, 1)
+    now = datetime.utcnow()
+    closed_discussion.preferences['landing_page'] = True
+    participant1_user.last_assembl_login = now
+    participant1_social_account.last_checked = now
+    path = test_participant1_webrequest.route_path(
+        'new_home', discussion_slug=closed_discussion.slug)
+    reply = test_app_participant1.get(path)
+    assert reply.status_code == 200
+    # unless my social login is expired
+    participant1_social_account.last_checked = long_ago
+    reply = test_app_participant1.get(path)
+    assert reply.status_code == 307
```

Comment:
Nitpicky: Instead of 307, or other numbers, use 

```
from pyramid.httpexceptions import HTTPTemporaryRedirect
HTTPTemporaryRedirect.status
```

Keeps tests super readable, and in the future, if the view is changed to throw another exception, it would be simple for other developers to know which exception the test should be replaced with.